synapse/handlers/sync.py

```diff
@@ -312,10 +317,11 @@ def ephemeral_by_room(self, sync_config, now_token, since_token=None):
         receipt_key = since_token.receipt_key if since_token else "0"
 
         receipt_source = self.event_sources.sources["receipt"]
-        receipts, receipt_key = yield receipt_source.get_new_events_for_user(
+        receipts, receipt_key = yield receipt_source.get_new_events(
             user=sync_config.user,
             from_key=receipt_key,
             limit=sync_config.filter.ephemeral_limit(),
+            room_ids=room_ids,
```

Comment:
It should. It does now. This implies that there is zero test coverage for this method.

Done